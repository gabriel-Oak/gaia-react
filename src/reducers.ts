import { combineReducers } from 'redux';

import authReducer from './pages/auth/authReducer';
import mainReducer from './pages/main/mainReducer';

export interface ReducersPool {
  mainReducer: object;
  authReducer: object;
}

export default combineReducers({
  authReducer,
  mainReducer
});