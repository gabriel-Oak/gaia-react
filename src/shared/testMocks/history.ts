
export const historyMock: historyTesting = {
  goBack: () => {},
  push: () => {}
};

export const matchMock = {
  isExact: true,
  params: {},
  path: '/',
  url: '/hgfd'
};

export interface historyTesting {
  goBack: Function;
  push: Function;
}