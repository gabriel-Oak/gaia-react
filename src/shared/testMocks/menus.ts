export const menusMock = [
  {
    data: '2019-11-04T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-05T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-06T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-07T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
  {
    data: '2019-11-08T22:16:00.000Z',
    guarnicao1: 'Batata Frita',
    guarnicao2: 'Ameixa da jamaica',
    opcao1: 'Omelete',
    opcao2: 'Bife',
    pratoPrincipal: 'Bacon',
    salada: 'Alface e tomate',
    sobremesa: 'Maça',
    __v: 0,
    _id: '5d08c85f4d6ea30b43bf6d3b'
  },
]