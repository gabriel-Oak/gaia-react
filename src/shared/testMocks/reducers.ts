import Action from "../../shared/interfaces/Action";
import { MainActionsTypes } from "../../pages/main/mainActionsTypes";

const MAIN_STATE = {
    title: 'Gaia',
    drawer: false,
    snackbar: {
        open: false,
    },
    user: {
        admin: true,
        email: 'test@email.cone',
        nome: 'Jorginho',
        user: 'string'
    },
}

export const mainReducerMock = (state = MAIN_STATE, action: Action) => {
    switch (action.type) {
        case MainActionsTypes.FETCH_USER:
            return { ...state, auth: true, user: action.value }
        default:
            return state;
    }
}
