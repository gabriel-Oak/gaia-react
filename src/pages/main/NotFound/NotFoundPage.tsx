import React, { PureComponent, ReactNode } from 'react'

import { setTitle } from '../mainActions';
import { ReducersPool } from '../../../reducers';

interface Props {
  setTitle: Function;
}

class NotFoundPage extends PureComponent<Props> {
  constructor(props: Props) {
    super(props)

    this.state = {

    }
  }

  render(): ReactNode {
    return (
      <main className="NotFound">
        Not found:
      </main>
    );
  }
}

const mapStateToProps = (state: ReducersPool) => {
  return {};
};

const mapDispatchToProps = {
  setTitle,
};

export default NotFoundPage
