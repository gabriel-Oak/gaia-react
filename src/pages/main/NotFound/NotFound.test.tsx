import React from 'react';

import { mount, shallow } from 'enzyme';

import { createStore, applyMiddleware } from 'redux';
import reducers from '../../../reducers';
import thunk from 'redux-thunk';
import NotFoundPage from './NotFoundPage';
import { matchMock, historyMock } from '../../../shared/testMocks/history';

describe('NotFound', () => {
  let store: any;
  let page: any;

  const redirect = jest.fn();

  beforeEach(() => {
    store = createStore(
      reducers,
      applyMiddleware(thunk)
    );

    historyMock.goBack = redirect;

    page = mount(
      <NotFoundPage
        match={matchMock}
        history={historyMock}
        store={store}
      />
    ).childAt(0);
  });

  it('Should render and match snapshot', () => {
    expect(page).toMatchSnapshot();
  });

  it('should redirect in button click', () => {
    page.find('button').simulate('click');
    expect(redirect.mock.calls.length).toBe(1);
  });
});