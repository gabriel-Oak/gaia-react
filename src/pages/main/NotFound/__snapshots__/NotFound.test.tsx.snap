// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Should render and match snapshot 1`] = `
<NotFoundPage
  history={
    Object {
      "goBack": [MockFunction],
      "push": [Function],
    }
  }
  match={
    Object {
      "isExact": true,
      "params": Object {},
      "path": "/",
      "url": "/hgfd",
    }
  }
  setTitle={[Function]}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <main
    className="NotFound Center-container"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <WithStyles(ForwardRef(Typography))
      variant="h1"
    >
      <ForwardRef(Typography)
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        variant="h1"
      >
        <h1
          className="MuiTypography-root MuiTypography-h1"
        >
           404 
        </h1>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Typography))
      gutterBottom={true}
      variant="h3"
    >
      <ForwardRef(Typography)
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
        gutterBottom={true}
        variant="h3"
      >
        <h3
          className="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom"
        >
          /hgfd
        </h3>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
      variant="contained"
    >
      <ForwardRef(Button)
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit",
            "contained": "MuiButton-contained",
            "containedPrimary": "MuiButton-containedPrimary",
            "containedSecondary": "MuiButton-containedSecondary",
            "containedSizeLarge": "MuiButton-containedSizeLarge",
            "containedSizeSmall": "MuiButton-containedSizeSmall",
            "disabled": "Mui-disabled",
            "endIcon": "MuiButton-endIcon",
            "focusVisible": "Mui-focusVisible",
            "fullWidth": "MuiButton-fullWidth",
            "iconSizeLarge": "MuiButton-iconSizeLarge",
            "iconSizeMedium": "MuiButton-iconSizeMedium",
            "iconSizeSmall": "MuiButton-iconSizeSmall",
            "label": "MuiButton-label",
            "outlined": "MuiButton-outlined",
            "outlinedPrimary": "MuiButton-outlinedPrimary",
            "outlinedSecondary": "MuiButton-outlinedSecondary",
            "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
            "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
            "root": "MuiButton-root",
            "sizeLarge": "MuiButton-sizeLarge",
            "sizeSmall": "MuiButton-sizeSmall",
            "startIcon": "MuiButton-startIcon",
            "text": "MuiButton-text",
            "textPrimary": "MuiButton-textPrimary",
            "textSecondary": "MuiButton-textSecondary",
            "textSizeLarge": "MuiButton-textSizeLarge",
            "textSizeSmall": "MuiButton-textSizeSmall",
          }
        }
        onClick={[Function]}
        variant="contained"
      >
        <WithStyles(ForwardRef(ButtonBase))
          className="MuiButton-root MuiButton-contained"
          component="button"
          disabled={false}
          focusRipple={true}
          focusVisibleClassName="Mui-focusVisible"
          onClick={[Function]}
          type="button"
        >
          <ForwardRef(ButtonBase)
            className="MuiButton-root MuiButton-contained"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            onClick={[Function]}
            type="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Voltar
              </span>
              <NoSsr>
                <WithStyles(undefined)
                  center={false}
                >
                  <ForwardRef(TouchRipple)
                    center={false}
                    classes={
                      Object {
                        "child": "MuiTouchRipple-child",
                        "childLeaving": "MuiTouchRipple-childLeaving",
                        "childPulsate": "MuiTouchRipple-childPulsate",
                        "ripple": "MuiTouchRipple-ripple",
                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                        "root": "MuiTouchRipple-root",
                      }
                    }
                  >
                    <span
                      className="MuiTouchRipple-root"
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component={null}
                        exit={true}
                      />
                    </span>
                  </ForwardRef(TouchRipple)>
                </WithStyles(undefined)>
              </NoSsr>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(Button)>
    </WithStyles(ForwardRef(Button))>
  </main>
</NotFoundPage>
`;
